File: RockitCompiler/benchmarks/bench_field_extract.go
package main

import (
	"fmt"
	"strings"
	"strconv"
)

func main() {
	parts := make([]string, 100)
	for i := 0; i < 100; i++ {
		parts[i] = "f" + strconv.Itoa(i)
	}
	line := strings.Join(parts, ",")

	result := ""
	for i := 0; i < 500000; i++ {
		fieldIdx := 0
		start := 0
		for j := 0; j < len(line); j++ {
			if line[j] == ',' {
				if fieldIdx == 50 {
					result = line[start:j]
					break
				}
				fieldIdx++
				start = j + 1
			}
		}
		if fieldIdx == 50 && result == "" {
			result = line[start:]
		}
	}
	fmt.Println(result)
}

File: RockitCompiler/benchmarks/bench_field_extract.js
const parts = [];
for (let i = 0; i < 100; i++) parts.push("f" + i);
const line = parts.join(",");

let result = "";
for (let iter = 0; iter < 500000; iter++) {
    let fieldIdx = 0;
    let start = 0;
    for (let j = 0; j < line.length; j++) {
        if (line[j] === ",") {
            if (fieldIdx === 50) {
                result = line.substring(start, j);
                break;
            }
            fieldIdx++;
            start = j + 1;
        }
    }
}
console.log(result);
